[Custom DC] Fix Disappearing Stat Var bug on Map and Scatter tool

juliawu opened the pull request:
This PR fixes some bugs in the map and scatter tool on Custom DC

## Bug 1: Disappearing stat vars

When a user selects a place + child place type in the dropdowns, the custom variables sometimes disappear from the sidebar. For example, when using the default OECD preview data, the OECD variables disappear from the sidebar when selecting "Asia" and "Country" in the dropdowns.

This happens because the custom variables don't have enough geographic coverage to meet the 10 place minimum set in PR #2314.

This PR sets this minimum to 1 place for custom DC (so that variables with no places are still filtered out) and leaves the 10 place minimum unchanged on main DC.

![Screenshot 2024-05-17 at 4 06 45 PM](https://github.com/datacommonsorg/website/assets/4034366/25b70c84-0e9d-44ce-9d12-cc05175641a0)

![Screenshot 2024-05-28 at 3 03 55 PM](https://github.com/datacommonsorg/website/assets/4034366/c4f83423-eca1-4126-87bc-c31d6ee460c8)

![Screenshot 2024-05-28 at 3 04 25 PM](https://github.com/datacommonsorg/website/assets/4034366/4e7b417e-6642-4b30-9758-a37e19a0894d)

![Screenshot 2024-05-28 at 3 04 50 PM](https://github.com/datacommonsorg/website/assets/4034366/ed98cb71-3f23-4c22-96be-ef053bb2aae2)


## Bug 2: ResizeObserver error

This PR also fixes a console error on the scatter tool encountered while debugging:
![Screenshot 2024-05-17 at 4 31 53 PM](https://github.com/datacommonsorg/website/assets/4034366/931de621-891f-4a78-a1a6-fed7d54e9f29)

dwnoble commented on server/templates/custom_dc/custom/base.html:
```diff
@@ -46,6 +46,9 @@
   <link href="{{ OVERRIDE_CSS_PATH }}" rel="stylesheet">
   {% endif %}
+  <script>
+    globalThis.isCustomDC = {{ config['CUSTOM']|int }};
```
this approach will require existing custom DC users to add this line in order to see the fix. is there a way we can push the fix to them requiring them to change any code?

juliawu replied:
Instead of setting a value in base.html, switched to using a config variable in app_env/*.py files, and setting globalThis on the tools pages themselves, instead of the template which could be overridden.

dwnoble commented on static/js/tools/map/stat_var_chooser.tsx:
```diff
@@ -114,10 +114,11 @@ export function StatVarChooser(props: StatVarChooserProps): JSX.Element {
-        NUM_ENTITIES_EXISTENCE,
-        samplePlaces.length
-      )}
+      numEntitiesExistence={
```
Can you extract this logic to a function and add some comments explaining the isCustomDC vs NUM_ENTITIES_EXISTENCE check

juliawu replied:
Good idea. Extracted to a helper function and added an explaining comment.

juliawu commented:
Thanks for the review! Because Custom DC now uses the new visualization tools by default, I also fixed this bug in the new visualization tools. Client (npm) tests have been updated to reflect the new default numEntitiesExistence = 1 when the flask config variable is not set.

beets reviewed (commented):
thanks for the fix (and sorry for the delay!)

it might be easier to reason about the setting if it were the default value to filter (1 for all tools, 10 for main dc in map and scatter). then the value can be easily overriden via config.

short of that, could we update the config var name from USE_STAT_VAR_FILTERING to FILTER_LOW_GEO_COVERAGE_STAT_VAR (or a more concise version of that)

beets reviewed (approved):
thanks for the update (and the concise flag name)

juliawu commented:
Thanks for the reviews!

For our records: took beet's advice and switched to setting the filter value in the config, and renamed the variable name to "MIN_STAT_VAR_GEO_COVERAGE"
